src/Tasks/RedistList.cs

```diff
@@ -159,7 +159,10 @@ internal int Count
         public bool IsFrameworkAssembly(string assemblyName)
         {
             AssemblyEntry entry = GetUnifiedAssemblyEntry(assemblyName);
-            if (entry != null && !String.IsNullOrEmpty(entry.RedistName))
+            AssemblyNameExtension assembly = new AssemblyNameExtension(assemblyName);
```

Comment:
I'm a little worried about perf here. AssemblyName is pretty slow, so creating a new `AssemblyNameExtension` every time could be an issue. 

Looking at the details of `GetUnifiedAssemblyEntry`, it always creates an `AssemblyNameExtension` anyway. What about return that as an overload with an out variable? Reusing it would eliminate extra overhead.

Reply:
Most of the cases, we won't hit the code in GetUnifiedAssemblyEntry that creates AssemblyNameExtension. If we really want to reuse it, we need to use a dictionary to hold the instances.

Reply:
Also, do we really call IsFrameworkAssembly a lot, just like calling GetUnifiedAssemblyEntry?